language-server/dm/rawDocument.ml

```diff
@@ -87,14 +87,20 @@ let range_of_loc raw loc =
   }
 
 let word_at_position raw pos : string option =
-  let r = Str.regexp {|\([a-zA-Z_][a-zA-Z_0-9]*\)|} in
-  let start = ref (loc_of_position raw pos) in
-  let word = ref None in
-  while (start.contents >= 0 && Str.string_match r raw.text start.contents) do
-    start := start.contents - 1;
-    word := Some (Str.matched_string raw.text);
-  done;
-  word.contents
+  try
+    let back_reg = Str.regexp {|[^a-zA-Z_0-9.']|} in
+    let start_ind = loc_of_position raw pos in
+    (* Search backwards until we find a character that cannot be part of a word *)
+    let first_non_word_ind = Str.search_backward back_reg raw.text start_ind in
+    let forward_reg = Str.regexp {|[^a-zA-Z_0-9']|} in
+    (* Search forwards ensuring that all characters are part of a well defined word. (Cannot start with [0-9'.] and cannot end with .)*)
+    let last_word_ind = Str.search_forward forward_reg raw.text start_ind in
+    let word = String.sub raw.text first_non_word_ind (last_word_ind - first_non_word_ind) in
```

Comment:
still looks like an off by one @rtetley .
Did we try `(Foo.bar` or the like?

Reply:
Nice catch, think #903 will fix